pkg/agent/openflow/feature_templates.go

```diff
@@ -0,0 +1,369 @@
+//                \                                      /        \                    /
+//              +------------------------------------------+    +--------------------------+
+//              |                 Feature                  |    |          Feature         |
+//              +------------------------------------------+    +--------------------------+
+//                                  |                    \                         /
+//                                  |                     \                       /
+//                                  |                      \                     /
+//                     +------------------------+        +-------------------------+
+//                     |        Pipeline        |        |         Pipeline        |
+//                     +------------------------+        +-------------------------+
+//
+// FeatureTable:
+// FeatureTable in Flexible Pipeline is like "iptables entry" in iptables. It is the basic unit to build OVS pipelines.
+//
+// Stage:
+// Stage in Flexible Pipeline is like "chain" in iptables. The order of Stages is fixed. A Stage can be empty or have
+// one or more FeatureTables. Note that, a FeatureTable cannot be shared by different Stages. All Stages include,
+//   - ClassifierStage
+//	 - ValidationStage
+//	 - ConntrackStateStage
+//	 - PreRoutingStage
+//	 - EgressSecurityStage
+//	 - RoutingStage
+//	 - PostRoutingStage
+//	 - SwitchingStage
+//	 - IngressSecurityStage
+//	 - ConntrackStage
+//	 - OutputStage
+//
+// Feature:
+// Feature in Flexible Pipeline is like "table" in iptables. A Feature is intended to implement a major function. Note
+// that, a Feature can have one or more Stages, and a Stage can be shared by multiple Features. All Features include,
+//   - PodConnectivity, implementation of connectivity for Pods
+//   - NetworkPolicy, implementation of K8s NetworkPolicy and Antrea NetworkPolicy
+//   - Service, implementation of K8s Service
+//   - Egress, implementation of Egress
+//   - Traceflow, implementation of Traceflow
+//   - Multicast, implementation of Multicast
+//
+// Pipeline:
+// This is a concept in Flexible Pipeline, not OVS pipeline, but a corresponding OVS pipeline will be realized in OVS
+// finally. A Pipeline is intended to implement a set of major functions. Note that, a Pipeline can have one or more
+// Features, but a Feature can be shared by multiple Pipelines. All Pipelines include,
+//   - Pipeline for IP
+//   - Pipeline for ARP
+//   - Pipeline for multicast
+//
+// How to insert a flow table with framework Flexible Pipeline?
+// - Select a target Pipeline, Feature and Stage.
+// - Define a FeatureTable with table name, Stage and priority. Arrange the FeatureTable in desired position with appropriate
+//   priority value. For example, there is a Stage A which has FeatureTable A1 (priority 100) and FeatureTable A2 (priority 200)
+//   declared by other Features. To insert a FeatureTable A3:
```

Comment:
s/other Features/another Feature/
FeatureTable can only be declared once, right?
Can we rename A1/A2/A3 to other names like Foo/Bar/Baz here? no strong opinion, but I feel it's easy for user to see them as A1->A2->A3

Reply:
Thanks, updated.